lib/internal/modules/cjs/loader.js

```diff
@@ -840,20 +844,25 @@ const CircularRequirePrototypeWarningProxy = new Proxy({}, {
   }
 });
 
-// Object.prototype and ObjectProtoype refer to our 'primordials' versions
+// Object.prototype and ObjectPrototype refer to our 'primordials' versions
 // and are not identical to the versions on the global object.
 const PublicObjectPrototype = global.Object.prototype;
 
 function getExportsForCircularRequire(module) {
-  if (module.exports &&
-      ObjectGetPrototypeOf(module.exports) === PublicObjectPrototype &&
-      // Exclude transpiled ES6 modules / TypeScript code because those may
-      // employ unusual patterns for accessing 'module.exports'. That should be
-      // okay because ES6 modules have a different approach to circular
-      // dependencies anyway.
-      !module.exports.__esModule) {
-    // This is later unset once the module is done loading.
-    ObjectSetPrototypeOf(module.exports, CircularRequirePrototypeWarningProxy);
+  if (module.exports) {
+    const proxy = getProxyDetails(module.exports);
+    // Exclude proxied modules that contain traps for the prototype functions.
+    if ((!proxy || (!proxy[1].setPrototypeOf && !proxy[1].getProxyDetails)) &&
```

Comment:
```suggestion
    if ((!proxy || (!proxy[1].setPrototypeOf && !proxy[1].getPrototypeOf)) &&
```

Reply:
I would skip the check for the individual proxy traps entirely. If somebody exports a Proxy, then we should just leave the object alone entirely.
```suggestion
    if (!proxy &&
```

Reply:
Looks good.